api/src/utils/open-payments.ts

```diff
@@ -0,0 +1,468 @@
+import {
+  type PendingGrant,
+  type WalletAddress,
+  type AuthenticatedClient,
+  type Quote,
+  isFinalizedGrant,
+  isPendingGrant,
+  createAuthenticatedClient
+} from '@interledger/open-payments'
+import {
+  createHeaders,
+  toWalletAddressUrl,
+  formatAmount,
+  timeout
+} from './utils.js'
+import { createId } from '@paralleldrive/cuid2'
+import type { Env } from '../index.js'
+
+export interface Amount {
+  value: string
+  assetCode: string
+  assetScale: number
+}
+
+export type QuoteResponse = Quote & { incomingPaymentGrantToken: string }
+
+type CreateIncomingPaymentParams = {
+  accessToken: string
+  walletAddress: WalletAddress
+  note?: string
+}
+
+type CheckPaymentResult =
+  | { success: true }
+  | {
+      success: false
+      error: { code: string; message: string; details?: Error }
+    }
+
+type QuoteGrantParams = {
+  authServer: string
+}
+
+type CreateOutgoingPaymentParams = {
+  walletAddress: WalletAddress
+  debitAmount: Amount
+  receiveAmount?: Amount
+  nonce?: string
+  paymentId: string
+}
+
+export class OpenPaymentsService {
+  private client: AuthenticatedClient | null = null
+  private static _instance: OpenPaymentsService
+  private readonly redirectUrl: string
+  private constructor(env: Env) {
+    this.redirectUrl = env.FRONTEND_URL
+  }
+
+  public static async getInstance(env: Env): Promise<OpenPaymentsService> {
+    if (!OpenPaymentsService._instance) {
+      OpenPaymentsService._instance = new OpenPaymentsService(env)
+      OpenPaymentsService._instance.client =
+        await OpenPaymentsService._instance.initClient(env)
+    }
+    return OpenPaymentsService._instance
+  }
+
+  private async initClient(env: Env): Promise<AuthenticatedClient> {

```

Comment:
This I believe is same as we've in `frontend`? Maybe this OP client stuff should be moved to a `shared/open-payments` package and reused in `frontend` and `api`?